tracing/honeycomb.go

```diff
@@ -1,34 +1,25 @@
 package tracing
 
 import (
-	"fmt"
-
-	"github.com/honeycombio/opentelemetry-exporter-go/honeycomb"
 	export "go.opentelemetry.io/otel/sdk/export/trace"
 )
 
 type Honeycomb struct {
-	APIKey      string `long:"honeycomb-api-key" description:"honeycomb.io api key"`
-	Dataset     string `long:"honeycomb-dataset" description:"honeycomb.io dataset name"`
-	ServiceName string `long:"honeycomb-service-name" description:"honeycomb.io service name" default:"concourse"`
+	APIKey  string `long:"honeycomb-api-key" description:"honeycomb.io api key"`
+	Dataset string `long:"honeycomb-dataset" description:"honeycomb.io dataset name"`
 }
 
 func (h Honeycomb) IsConfigured() bool {
 	return h.APIKey != "" && h.Dataset != ""
 }
 
-func (h Honeycomb) Exporter() (export.SpanSyncer, error) {
-	exporter, err := honeycomb.NewExporter(
-		honeycomb.Config{
-			APIKey: h.APIKey,
+func (h Honeycomb) Exporter() (export.SpanExporter, error) {
+	return OTLP{
+		Address: "api.honeycomb.io:443",
+		Headers: map[string]string{
+			"x-honeycomb-team":    h.APIKey,
+			"x-honeycomb-dataset": h.Dataset,
 		},
-		honeycomb.TargetingDataset(h.Dataset),
-		honeycomb.WithServiceName(h.ServiceName),
```

Comment:
Yeah I noticed that - I made a note of that in the commit message:

```
...

this means we no longer can set CONCOURSE_TRACING_HONEYCOMB_SERVICE_NAME
- however, we can instead set CONCOURSE_TRACING_SERVICE_NAME, which
should do the same thing effectively
```

We set the conventional service name field (which is `service.name`, rather than `service_name`) regardless of what exporter is configured: https://github.com/concourse/concourse/blob/f0478ad034157cd34cee71cce4040ac7ce6611ad/tracing/tracer.go#L38

I wonder if there's really a need to have both? And if not, it's probably better to go with the standard convention I figure. This does mean this is a breaking change, though.

@spire-allyjweir I know you're the one who contributed honeycomb support to begin with - does this change seem reasonable to you?

Reply:
Honeycomb gives special meaning to `service_name` and it's used in the UI. Not sure why it's done this way, but it will result in a different behaviour

![Screen Shot 2021-04-08 at 3 05 42 PM](https://user-images.githubusercontent.com/17577073/114082658-ff815f00-987b-11eb-919c-7fe9ef8b7a7a.png)

When only `service.name` is used, it shows up as undefined and analytics is lost
![Screen Shot 2021-04-08 at 3 10 42 PM](https://user-images.githubusercontent.com/17577073/114083118-9cdc9300-987c-11eb-9ae9-b01583140a94.png)

Reply:
Hm I'm not seeing that - using the following docker-compose override:

```yaml
version: '3'

services:
  web:
    environment:
      CONCOURSE_TRACING_HONEYCOMB_API_KEY: xxxxx
      CONCOURSE_TRACING_HONEYCOMB_DATASET: test-concourse
      CONCOURSE_TRACING_SERVICE_NAME: parent-service
```

I'm able to see the service name being set properly in Honeycomb:
<img width="618" alt="Screen Shot 2021-04-14 at 11 29 33 AM" src="https://user-images.githubusercontent.com/26583442/114737319-e7468f80-9d14-11eb-9a00-cc129ace27cd.png">

It even seems to ascribe special meaning to the property elsewhere in the UI:
<img width="903" alt="Screen Shot 2021-04-14 at 11 29 45 AM" src="https://user-images.githubusercontent.com/26583442/114737394-f88f9c00-9d14-11eb-9ecb-869f6044ce74.png">

Reply:
huh, I just tested it again and it works as you said. I must've been looking at the wrong timespan or something.